framework/src/main/java/org/fulib/fx/controller/internal/ReflectionSidecar.java

```diff
@@ -456,14 +456,16 @@ private EventHandler<KeyEvent> createKeyEventHandler(Method method, Object insta
         };
     }
 
-    private boolean keyEventMatchesAnnotation(KeyEvent event, OnKey annotation) {
-        return (annotation.code() == KeyCode.UNDEFINED || event.getCode() == annotation.code()) &&
-            (annotation.character().isEmpty() || event.getCharacter().equals(annotation.character())) &&
-            (annotation.text().isEmpty() || event.getText().equals(annotation.text())) &&
-            (event.isShiftDown() || !annotation.shift()) &&
-            (event.isControlDown() || !annotation.control()) &&
-            (event.isAltDown() || !annotation.alt()) &&
-            (event.isMetaDown() || !annotation.meta());
+    private boolean keyEventMatchesAnnotation(KeyEvent event, OnKey onKey) {
+        return
+            (onKey.code() == KeyCode.UNDEFINED || event.getCode() == onKey.code()) &&
+            (onKey.character().isEmpty() || event.getCharacter().equals(onKey.character())) &&
+            (onKey.text().isEmpty() || event.getText().equals(onKey.text())) &&
+
+            (((onKey.shift() && event.isShiftDown())) || (!onKey.shift() && !event.isShiftDown())) &&
+            (((onKey.control() && event.isControlDown())) || (!onKey.control() && !event.isControlDown())) &&
+            (((onKey.alt() && event.isAltDown())) || (!onKey.alt() && !event.isAltDown())) &&
+            (((onKey.meta() && event.isMetaDown())) || (!onKey.meta() && !event.isMetaDown()));
```

Comment:
This does not use the `strict()` method ;) The expressions are equivalent to `onKey.shift() == event.isShiftDown()`, so it is always strict.

Reply:
I fixed it.